src/ServiceResolver/Resolvers/Discoverer/DiscoveredDefinition.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+namespace Bauhaus\ServiceResolver\Resolvers\Discoverer;
+
+use Bauhaus\ServiceResolver\ActualDefinition;
+use Bauhaus\ServiceResolver\Definition;
+use Psr\Container\ContainerInterface as PsrContainer;
+use ReflectionClass as RClass;
+use ReflectionParameter as RParam;
+
+/**
+ * @internal
+ */
+final class DiscoveredDefinition implements Definition
```

Comment:
I agree autowire is a world already in use but I disagree with using this word here.

Originally the autowire term is used when dependencies are automatically injected if they were not explicitly set, and that is why the word `wire` was used since it is basically connecting the services (Behat/Behat#1071).

It is true that, in order to achieve service discovering, it is indeed needed to automatically inject dependencies, but there is an important difference here if you compare it to the original autowire propose. That I believe the term discover suits better in this case.

Reply:
I agree. _Wiring_ implies a relation between two (or more) things, and here you're only establishing one side of the relationship.